pkg/ca/fileca/fileca.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package fileca
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/ca/x509ca"
+	"github.com/sigstore/fulcio/pkg/challenges"
+)
+
+type fileCA struct {
+	sync.RWMutex
+
+	cert *x509.Certificate
+	key  crypto.Signer
+}
+
+// NewFileCA returns a file backed certificate authority. Expects paths to a
+// certificate and key that are PEM encoded and not password protected.
+func NewFileCA(certPath, keyPath string, watch bool) (ca.CertificateAuthority, error) {
```

Comment:
Do you think it'd be better to have the key and certificate in the same file? This would prevent cases where one gets updated and not the other.

Reply:
If these come from e.g. a secret volume mount, then they'd very likely come in from separate keys as separate files.

Reply:
I think the pattern of a seperate key and cert file is really well established so best to follow. If we put them in one file when need to dream up some unique-to-fulcio format for that. Do we just append the cert pem and encrypted key? It will be something we have to explain away exit.

For what its worth, the case in which one is updated and other isn't is covered. The lack of matching key-pair is noticed and we don't update the existing key-pair. Once the cert and key match they are loaded and updated. See https://github.com/sigstore/fulcio/pull/280/files#diff-8e2eea76cde95683666df85d24ad78e87b4f6529d77b14f27ed884a3fd930559R30-R33